### ISSUE TITLE
A problem about Zero-padding

### ISSUE THREAD

**Issue by zechengtang** [cnn-mind-gap/cnn-mind-gap.github.io#2]
It was nice to read your great paper, but I have a question about one sentence in the paper: 'SAME 0-padding is by far the most widely used method. It can make training and inference 50% faster compared to other methods.' . Why is that so?

**Comment by bilalsal** [comment]
Hi @newcomertzc,
thank you for reaching out.

We trained various ResNet models on ImageNet under three padding methods available in PyTorch ("zeros", "reflect", "circular").
On a single GPU node, ResNet-18 took less than 3 days under zero padding (fastest), and about 4.5 days under circular padding (slowest). Similar gap applies when using deeper ResNet.

The reason behind the gap is that non-zero padding methods are applied by slicing the boundary into thin tensors (e.g. X[:1, :], X[-1:, :], X[:, :1], X[:, -1:], assuming a padding of (1, 1, 1, 1). These tensors are stitched back together with X using `torch.cat` into one tensor that constitutes the padded input to the conv layer.
These operations are done for each conv layer and are implemented in Python (not in C++).
Accordingly, they incur a significant overhead for every forward pass (note that the backward pass will also differentiate through the padding area). 
With zero padding, there are no tensor operations needed and the backward pass is simplified.

Hope this helps

**Comment by zechengtang** [comment]
Thank you for your answer!
